src/main/java/com/philips/research/bombar/core/domain/licenses/ObligationsAnalyzer.java

```diff
@@ -0,0 +1,57 @@
+package com.philips.research.bombar.core.domain.licenses;
+
+import com.philips.research.bombar.core.domain.Dependency;
+import com.philips.research.bombar.core.domain.Project;
+import com.philips.research.bombar.core.domain.Relation;
+import pl.tlinkowski.annotation.basic.NullOr;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+public class ObligationsAnalyzer {
+    private final LicenseRegistry registry;
+    private final Project project;
+
+    public ObligationsAnalyzer(LicenseRegistry registry, Project project) {
+        this.registry = registry;
+        this.project = project;
+    }
+
+    /**
+     * @return map of license obligations mapped to the present dependencies including the license
+     */
+    public Map<String, Set<Dependency>> findObligations() {
+        Map<String, Set<Dependency>> obligations = new HashMap<>();
+        project.getDependencies().forEach(dep -> {
+            final var relationship = dep.getStrongestUsage().orElse(Relation.Relationship.weakest());
+            final var distribution = project.getDistribution();
```

Comment:
This variable is now again defined for every dependency, although it is not related to the dependency.

My suggestion is to move the variable outside of the `forEach(...)` iteration.